### ISSUE TITLE
Allow access to raw value used as Foreign Key in a relation

### ISSUE THREAD

**Issue by robmurtagh** [prisma/prisma2#608]
The following is valid to Photon, but not to Lift:

```
model Foo {
  bar       Bar         @map("bar_uuid")
  barId     String      @map("bar_uuid")
}
```

If would be fantastic if either the above was allowed, or if `barId` could be autogenerated on the type `Foo`.

Making `barId` accessible significantly cleans the syntax for callers (and I think without `barId` the query engine may even implement joins or something similar under the covers).

For context because we are struggling with some Photon performance issues, we're using some manual workarounds. This backs out to us having many cases like `barId` in our schema but commenting and uncommenting the `barId` line depending on whether we're using `Photon`, or `Lift`.

**Comment by johannesschobel** [comment]
Dear @robmurtagh ,

this would allow also to call something like:
```ts
const posts = await this.prisma.posts.findMany( { where: { author: author.id } } );
```
right?

Currently, this query does not work for me, IF the `author` field is a (1:n) relationship (_one author has many post, one post belongs to one author_).

If i traverse the entire datamodel, like:
```ts
const posts = await this.prisma.authors.findOne( { where: { id: author.id } } ).posts();
```
it works. However, the second option may be significantly more complex, depending on your datamodel and query-paths..

**Comment by johannesschobel** [comment]
Regarding my issue described above..
I just found out, that you can do something like:
```ts
const posts = await this.prisma.posts.findMany({
  where: {
    author: { id: account.id }
  }
});
```

Hope this helps someone

**Comment by robmurtagh** [comment]
Yes, the second syntax is the right way to do this, and works fine for use cases like that. I'm not sure what the performance implications of it are right now (e.g. whether it uses a join or something like this).

What can be tricky though, is once you have broken out of the fluent API. It would often be nice when passing around a `post` object to utilise its `authorId` (e.g. to pass to other HTTP services). However that wouldn't be possible in the current version of Lift/Photon.

**Comment by johannesschobel** [comment]
yeah, but if you use the "fluent api" you are constantly writing the "base part" all over again.. and that may be subject to errors..

Currently, it's not possible to return a "fluent" type, where you then can chain onto, right?

All the best
